src/test/java/org/openlmis/stockmanagement/controller/StockCardControllerTest.java

```diff
@@ -172,14 +186,79 @@ public void shouldSucceedWithValidAdjustment() {
     when(facilityRepository.getById(fId)).thenReturn(defaultFacility);
     when(productService.getById(pId)).thenReturn(defaultProduct);
     when(stockAdjustmentReasonRepository.getAdjustmentReasonByName(reasonName)).thenReturn(reason);
-    when(service.getOrCreateStockCard(fId, pId)).thenReturn(dummyCard);
-    when(lotRepository.getLotOnHandByStockCardAndLot(dummyCard.getId(), lotId)).thenReturn(null);
+    when(stockCardService.getOrCreateStockCard(fId, pId)).thenReturn(dummyCard);
+    when(lotRepository.getLotOnHandByStockCardAndLot(eq(dummyCard.getId()), any(Long.class))).thenReturn(null);
     ResponseEntity response = controller.processStock(fId, Collections.singletonList(event), request);
 
     // verify
     StockCardEntry entry = new StockCardEntry(dummyCard, StockCardEntryType.ADJUSTMENT, event.getQuantity() * -1);
     entry.setAdjustmentReason(reason);
-    verify(service).addStockCardEntries(Collections.singletonList(entry));
+    verify(stockCardService).addStockCardEntries(Collections.singletonList(entry));
     assertThat(response.getStatusCode(), is(HttpStatus.OK));
   }
+
+
+  @Test
+  public void shouldOnlyReturnEmptyLotsWhenRequested()
```

Comment:
I had considered doing so, and wonder if this is a matter of preference. I don't think this is what "Effective Unit Testing" would call a "Split Personality" test, though. Would certainly be interested in discussing.